edb/server/protocol/args_ser.pyx

```diff
@@ -209,6 +205,55 @@ cdef WriteBuffer recode_bind_args(
     return out_buf
 
 
+cdef WriteBuffer recode_array(
+    dbv: dbview.DatabaseConnectionView,
+    FRBuffer* in_buf,
+    out_buf: WriteBuffer,
+    in_len: ssize_t,
+    array_tid: int32_t,
+):
+    # For a standalone array, we still need to inject oids and reject
+    # NULL elements.
+    cdef:
+        WriteBuffer buf
+        ssize_t cnt
+        ssize_t idx
+        ssize_t num
+        ssize_t tag
+        FRBuffer sub_buf
+
+    frb_slice_from(&sub_buf, in_buf, in_len)
+
+    val = hton.unpack_int32(frb_read(&sub_buf, 4)) # ndims
+    if val != 1 and val != 0:
+        raise errors.QueryError("unsupported array dimensions")
```

Comment:
`InputDataError`?

Reply:
Sure. Made all the errors in args_ser.pyx be InputDataError